OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs

```diff
@@ -62,6 +62,12 @@ public static CVec AdjustForBuildingSize(BuildingInfo buildingInfo)
 			return new CVec(dim.X / 2, dim.Y > 1 ? (dim.Y + 1) / 2 : 0);
 		}
 
+		public static CVec AccountForBuidingSize(BuildingInfo buildingInfo)
```

Comment:
This is the cause of your AFLD placement issues, and there's also a typo in the method name.

Please remove this, and use AdjustBuildingSize.

```diff
diff --git a/OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs b/OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs
index 5fe0d5e..172fe00 100644
--- a/OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs
+++ b/OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs
@@ -62,12 +62,6 @@ public static CVec AdjustForBuildingSize(BuildingInfo buildingInfo)
 			return new CVec(dim.X / 2, dim.Y > 1 ? (dim.Y + 1) / 2 : 0);
 		}
 
-		public static CVec AccountForBuidingSize(BuildingInfo buildingInfo)
-		{
-			var dim = buildingInfo.Dimensions;
-			return new CVec(dim.X / 2, dim.Y % 2 == 0 ? (dim.Y + 1) / 2 : 0);
-		}
-
 		public static WVec CenterOffset(World w, BuildingInfo buildingInfo)
 		{
 			var dim = buildingInfo.Dimensions;
diff --git a/OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorLogic.cs b/OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorLogic.cs
index 0cd4b46..39d6cd8 100644
--- a/OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorLogic.cs
+++ b/OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorLogic.cs
@@ -108,7 +108,7 @@ public MapEditorLogic(Widget widget, World world, WorldRenderer worldRenderer)
 						var buildingInfo = traits.GetOrDefault<BuildingInfo>();
 						if (buildingInfo != null)
 						{
-							cell -= FootprintUtils.AccountForBuidingSize(buildingInfo) / 2;
+							cell -= FootprintUtils.AdjustForBuildingSize(buildingInfo);
 							offset = FootprintUtils.CenterOffset(world, buildingInfo);
 						}
 					}
diff --git a/OpenRA.Mods.Common/Widgets/MapEditorWidget.cs b/OpenRA.Mods.Common/Widgets/MapEditorWidget.cs
index 05d3b9a..f28395a 100644
--- a/OpenRA.Mods.Common/Widgets/MapEditorWidget.cs
+++ b/OpenRA.Mods.Common/Widgets/MapEditorWidget.cs
@@ -189,12 +189,17 @@ public override bool HandleMouseInput(MouseInput mi)
 							var newActorReference = new ActorReference(SelectedActor);
 							newActorReference.Add(new OwnerInit(SelectedOwner));
 
-							newActorReference.Add(new LocationInit(MouseCursorCell));
+							var xy = MouseCursorCell;
 							var info = rules.Actors[SelectedActor];
+							var buildingInfo = info.Traits.GetOrDefault<BuildingInfo>();
+							if (buildingInfo != null)
+								xy -= FootprintUtils.AdjustForBuildingSize(buildingInfo);
+
+							newActorReference.Add(new LocationInit(xy));
 							var mobile = info.Traits.GetOrDefault<MobileInfo>();
 							if (mobile != null && mobile.SharesCell)
 							{
-								var subcell = world.ActorMap.FreeSubCell(MouseCursorCell);
+								var subcell = world.ActorMap.FreeSubCell(xy);
 								if (subcell != SubCell.Invalid)
 									newActorReference.Add(new SubCellInit(subcell));
 							}
```